apollo-ios-pagination/Sources/ApolloPagination/AsyncGraphQLQueryPager.swift

```diff
@@ -248,25 +178,12 @@ public class AsyncGraphQLQueryPager<Model>: Publisher {
 
   public func receive<S>(
     subscriber: S
-  ) where S: Subscriber, Never == S.Failure, Result<(Model, UpdateSource), any Error> == S.Input {
-    publisher.subscribe(subscriber)
-  }
-}
-
-extension AsyncGraphQLQueryPager: Equatable where Model: Equatable {
-  public static func == (lhs: AsyncGraphQLQueryPager<Model>, rhs: AsyncGraphQLQueryPager<Model>) -> Bool {
-    let left = lhs._subject.value
-    let right = rhs._subject.value
-
-    switch (left, right) {
-    case (.success((let leftValue, let leftSource)), .success((let rightValue, let rightSource))):
-      return leftValue == rightValue && leftSource == rightSource
-    case (.failure(let leftError), .failure(let rightError)):
-      return leftError.localizedDescription == rightError.localizedDescription
-    case (.none, .none):
-      return true
-    default:
-      return false
-    }
+  ) where S: Subscriber, Never == S.Failure, Result<Model, any Error> == S.Input {
+    publisher
+      .removeDuplicates(by: { _lhs, _rhs in
+        guard let lhs = try? _lhs.get(), let rhs = try? _rhs.get() else { return false }
+        return lhs == rhs
+      })
```

Comment:
Is this the only reason we added the requirement to make models `Equatable`? I'd prefer if we didn't add restrictions that aren't 100% necessary.

Further, I'm not convinced we should even be calling `removeDuplicates` here. Is there a situation caused by this PR where we are now anticipating duplicated updates that we didn't not before?